VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentCancelReschedule.tsx

```diff
@@ -0,0 +1,277 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { Button, ButtonVariants } from '@department-of-veterans-affairs/mobile-component-library'
+import { UseMutateFunction } from '@tanstack/react-query'
+import { TFunction } from 'i18next'
+
+import { AppointmentAttributes } from 'api/types'
+import { Box, ClickToCallPhoneNumber, LinkWithAnalytics, TextView } from 'components'
+import { Events } from 'constants/analytics'
+import { NAMESPACE } from 'constants/namespaces'
+import { AppDispatch } from 'store'
+import { VATheme } from 'styles/theme'
+import { a11yLabelVA } from 'utils/a11yLabel'
+import { logAnalyticsEvent } from 'utils/analytics'
+import {
+  AppointmentDetailsScreenType,
+  AppointmentDetailsSubType,
+  AppointmentDetailsSubTypeConstants,
+  AppointmentDetailsTypeConstants,
+  getAppointmentAnalyticsDays,
+  getAppointmentAnalyticsStatus,
+} from 'utils/appointments'
+import { showSnackBar } from 'utils/common'
+import getEnv from 'utils/env'
+import { useAppDispatch, useDestructiveActionSheet, useDestructiveActionSheetProps, useTheme } from 'utils/hooks'
+
+const { LINK_URL_VA_SCHEDULING, WEBVIEW_URL_FACILITY_LOCATOR } = getEnv()
+
+type AppointmentCancelRescheduleProps = {
+  appointmentID: string
+  attributes: AppointmentAttributes
+  subType: AppointmentDetailsSubType
+  type: AppointmentDetailsScreenType
+  goBack: () => void
+  cancelAppointment?: UseMutateFunction<unknown, Error, string, unknown>
+}
+
+const cancelButton = (
+  pendingAppointment: boolean,
+  appointmentID: string,
+  attributes: AppointmentAttributes,
+  goBack: () => void,
+  t: TFunction,
+  theme: VATheme,
+  dispatch: AppDispatch,
+  confirmAlert: (props: useDestructiveActionSheetProps) => void,
+  cancelId?: string,
+  cancelAppointment?: UseMutateFunction<unknown, Error, string, unknown>,
+) => {
+  const onPress = () => {
+    logAnalyticsEvent(
+      Events.vama_apt_cancel_clicks(
```

Comment:
updated the naming of the variable. I left the name field the same since this analytic carried over and didn't want to end up having 2 analytics for the same thing between versions.